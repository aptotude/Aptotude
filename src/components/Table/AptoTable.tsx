import React from 'react';
import { StandardTypes } from '../../utils/standardTypes';
import classNames from 'classnames';
import './aptoTable.scss';

const COMPONENT_PREFIX = 'AptoTable';

export interface AptoTableDisplayProps extends StandardTypes {
  striped?: boolean | undefined;
  bordered?: boolean | undefined;
  borderless?: boolean | undefined;
  hover?: boolean | undefined;
  small?: boolean | undefined;
  responsive?: boolean | undefined;
}

export class AptoTable extends React.PureComponent<AptoTableDisplayProps> {
  public static defaultProps = {
    striped: false,
    bordered: false,
    borderless: false,
    hover: false,
    small: false,
    responsive: false
  };
  public render() {
    const {
      children,
      striped,
      bordered,
      borderless,
      small,
      hover,
      responsive,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      striped && `${COMPONENT_PREFIX}--striped`,
      bordered && `${COMPONENT_PREFIX}--bordered`,
      borderless && `${COMPONENT_PREFIX}--borderless`,
      hover && `${COMPONENT_PREFIX}--hover`,
      small && `${COMPONENT_PREFIX}--small`
    );

    const tableDom = (
      <table className={classes} {...rest}>
        {children}
      </table>
    );

    if (responsive) {
      return (
        <div className={`${COMPONENT_PREFIX}--responsive`}>{tableDom}</div>
      );
    }

    return tableDom;
  }
}
