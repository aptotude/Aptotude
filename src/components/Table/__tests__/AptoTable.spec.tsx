import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoTable } from '../AptoTable';

afterEach(() => {
  cleanup();
});

describe('AptoTable', () => {
  it('Renders Component', () => {
    const { container } = render(
      <AptoTable>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders extra classes', () => {
    const { container } = render(
      <AptoTable className="foo">
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders striped', () => {
    const { container } = render(
      <AptoTable striped>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders bordered', () => {
    const { container } = render(
      <AptoTable bordered>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders borderless', () => {
    const { container } = render(
      <AptoTable borderless>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders small', () => {
    const { container } = render(
      <AptoTable small>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders hover', () => {
    const { container } = render(
      <AptoTable hover>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders responsive', () => {
    const { container } = render(
      <AptoTable responsive>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </AptoTable>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
