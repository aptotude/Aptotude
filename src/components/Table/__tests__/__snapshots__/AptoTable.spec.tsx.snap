// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoTable Renders Component 1`] = `
<table
  class="AptoTable"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTable Renders bordered 1`] = `
<table
  class="AptoTable AptoTable--bordered"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTable Renders borderless 1`] = `
<table
  class="AptoTable AptoTable--borderless"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTable Renders extra classes 1`] = `
<table
  class="foo"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTable Renders hover 1`] = `
<table
  class="AptoTable AptoTable--hover"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTable Renders responsive 1`] = `
<div
  class="AptoTable--responsive"
>
  <table
    class="AptoTable"
  >
    <tbody>
      <tr>
        <td>
          hi
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`AptoTable Renders small 1`] = `
<table
  class="AptoTable AptoTable--small"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTable Renders striped 1`] = `
<table
  class="AptoTable AptoTable--striped"
>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;
